File: src/SQLAgent/Entities/AIProvider.cs
namespace SQLAgent.Entities;

/// <summary>
/// AI 提供商类型
/// </summary>
public enum AIProviderType
{
    /// <summary>
    /// OpenAI 官方端点
    /// </summary>
    OpenAI,
    
    /// <summary>
    /// Azure OpenAI 服务
    /// </summary>
    AzureOpenAI,
    
    /// <summary>
    /// OpenAI 兼容的自定义端点
    /// </summary>
    CustomOpenAI,
    
    /// <summary>
    /// Ollama 本地模型
    /// </summary>
    Ollama
}

/// <summary>
/// AI 提供商配置
/// </summary>
public record AIProvider
{
    /// <summary>
    /// 唯一标识
    /// </summary>
    public string Id { get; init; } = Guid.NewGuid().ToString();
    
    /// <summary>
    /// 提供商名称
    /// </summary>
    public string Name { get; init; } = string.Empty;
    
    /// <summary>
    /// 提供商类型
    /// </summary>
    public AIProviderType Type { get; init; }
    
    /// <summary>
    /// API 端点（可选，OpenAI 使用默认端点）
    /// </summary>
    public string? Endpoint { get; init; }
    
    /// <summary>
    /// API 密钥
    /// </summary>
    public string ApiKey { get; init; } = string.Empty;
    
    /// <summary>
    /// 可用的模型列表（逗号分隔）
    /// </summary>
    public string AvailableModels { get; init; } = string.Empty;
    
    /// <summary>
    /// 默认模型
    /// </summary>
    public string? DefaultModel { get; init; }
    
    /// <summary>
    /// 是否启用
    /// </summary>
    public bool IsEnabled { get; init; } = true;
    
    /// <summary>
    /// 附加配置（JSON 格式）
    /// </summary>
    public string? ExtraConfig { get; init; }
    
    /// <summary>
    /// 创建时间
    /// </summary>
    public DateTime CreatedAt { get; init; } = DateTime.UtcNow;
    
    /// <summary>
    /// 最后更新时间
    /// </summary>
    public DateTime UpdatedAt { get; init; } = DateTime.UtcNow;
}

File: web/src/types/aiProvider.ts
/**
 * AI 提供商类型
 */
export type AIProviderType = "OpenAI" | "AzureOpenAI" | "CustomOpenAI" | "Ollama";

export const AIProviderTypes = {
  OpenAI: "OpenAI" as const,
  AzureOpenAI: "AzureOpenAI" as const,
  CustomOpenAI: "CustomOpenAI" as const,
  Ollama: "Ollama" as const,
};

/**
 * AI 提供商
 */
export interface AIProvider {
  id: string;
  name: string;
  type: AIProviderType;
  endpoint?: string;
  apiKey: string;
  availableModels: string[];
  defaultModel?: string;
  isEnabled: boolean;
  extraConfig?: string;
  createdAt: string;
  updatedAt: string;
}

/**
 * AI 提供商输入（创建/更新）
 */
export interface AIProviderInput {
  name: string;
  type: string;
  endpoint?: string;
  apiKey: string;
  availableModels: string | string[]; // 逗号分隔
  defaultModel?: string;
  isEnabled: boolean;
  extraConfig?: string;
}

/**
 * AI 模型信息
 */
export interface AIModel {
  name: string;
  displayName: string;
  isDefault: boolean;
}

/**
 * 常用的模型列表
 */
export const COMMON_MODELS = {
  OpenAI: ["gpt-4.1", "gpt-4.1-mini", "gpt-5", "gpt-5-mini"],
  AzureOpenAI: ["gpt-4.1", "gpt-4.1-mini", "gpt-5", "gpt-5-mini"],
  CustomOpenAI: ["gpt-4.1", "gpt-4.1-mini", "gpt-5", "gpt-5-mini"],
  Ollama: ["gpt-oss:20b-cloud"],
};
